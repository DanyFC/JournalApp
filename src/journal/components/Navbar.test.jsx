/**
 * @jest-environment jsdom
*/
import { fireEvent, render, screen } from '@testing-library/react';
import Navbar from './Navbar';

/* eslint-disable no-undef */
const mockedStartLogout = jest.fn()

jest.mock('./../../store/auth/thunks', () => ({
  startLogout: () => mockedStartLogout
}))

jest.mock('react-redux', () => ({
  useDispatch: () => (fn) => fn()
}))

describe('Tests in <Navbar />', () => {

  test('Can be match with snapshot. ', () => {
    const { container } = render(<Navbar drawerWidth={200} />)

    expect(container).toMatchSnapshot()
  })

  test('Should call startLogout when logout button is clicked.', () => {
    render(<Navbar />)

    const logoutButton = screen.getByLabelText('logout-btn')
    fireEvent.click(logoutButton)

    expect(mockedStartLogout).toHaveBeenCalled()
  })
})
/* eslint-enable no-undef */
