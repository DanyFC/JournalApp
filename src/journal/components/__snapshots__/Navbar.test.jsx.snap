// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Navbar /> Can be match with snapshot.  1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1a7a1ce-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
        >
          Journal App
        </div>
        <button
          aria-label="logout-btn"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LogoutOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17 8-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
